src/main/jbake/content/blog/2017/07/opentracing-ejb.adoc

```diff
@@ -0,0 +1,393 @@
+for a complete tracing picture, without any explicit tracing code, apart from passing the context around.
+
+`src/main/java/io/opentracing/contrib/ejb/example/InventoryService.java`
+```java
+package io.opentracing.contrib.ejb.example;
+
+import io.opentracing.SpanContext;
+import io.opentracing.contrib.ejb.OpenTracingInterceptor;
+
+import javax.ejb.Asynchronous;
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+import javax.interceptor.Interceptors;
+import java.util.logging.Logger;
+
+/**
+ * This is an asynchronous stateless EJB with spans created automatically by the interceptor. Note that the span context
+ * that this method sees is <b>not</b> the same as the span context sent by the caller: the interceptor wraps this
+ * method call on its own span, and replaces the span context by the context of this new span. This is done so that this
+ * span context can be passed along to the next service "as is".
+ *
+ * @author Juraci Paixão Kröhling
+ */
+@Asynchronous
+@Stateless
+@Interceptors({OpenTracingInterceptor.class})
+public class InventoryService {
+    private static final Logger log = Logger.getLogger(InventoryService.class.getName());
+
+    @Inject
+    InventoryNotificationService inventoryNotificationService;
+
+    public void changeInventory(SpanContext context) {
+        log.info("Changing the inventory");
+        inventoryNotificationService.sendNotification(context);
+    }
+}
+```
+
+And finally, our last service, `InventoryNotificationService`: on this case, we notify another set of backend systems
+that a new order has been placed. Again, this is an asynchronous EJB and works like the one above, but additionally,
+we wanted to manually create a "business span", called `sendNotification`. This method could send several notifications,
+wrapping each one into a span of its own. As we manually started it, we manually finish it as well.
+
+`src/main/java/io/opentracing/contrib/ejb/example/InventoryNotificationService.java`
+```java
+package io.opentracing.contrib.ejb.example;
+
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+import io.opentracing.util.GlobalTracer;
+
+import javax.ejb.Asynchronous;
+import javax.ejb.Stateless;
+import java.util.logging.Logger;
+
+/**
+ * This is the final call in the chain. This is an asynchronous stateless EJB, which obtains the span context
+ * via a method parameter. This bean is not intercepted in any way by us, so, the span context received is exactly
+ * the same as what was sent by the caller.
+ *
+ * @author Juraci Paixão Kröhling
+ */
+@Stateless
+@Asynchronous
+public class InventoryNotificationService {
+    private static final Logger log = Logger.getLogger(InventoryNotificationService.class.getName());
+
+    public void sendNotification(SpanContext context) {
+        Span span = GlobalTracer.get().buildSpan("sendNotification").asChildOf(context).startManual();
```

Comment:
Wondering whether it would be better to @Inject Tracer - similar to what is likely to be used in microprofile.io.